rocketpy/mathutils/function.py

```diff
@@ -333,142 +345,173 @@ def set_extrapolation(self, method="constant"):
         self : Function
             The Function object.
         """
-        self.__extrapolation__ = method
-        self.set_get_value_opt()
+        if not callable(self.source):
+            self.__extrapolation__ = self._validate_extrapolation(method)
+            self._set_extrapolation_func()
         return self
 
-    def set_get_value_opt(self):
-        """Crates a method that evaluates interpolations rather quickly
-        when compared to other options available, such as just calling
-        the object instance or calling ``Function.get_value`` directly. See
-        ``Function.get_value_opt`` for documentation.
+    def _set_interpolation_func(self):
+        """Defines interpolation function used by the Function. Each
+        interpolation method has its own function with exception of shepard,
+        which has its interpolation/extrapolation function defined in
+        ``Function.__interpolate_shepard__``. The function is stored in
+        the attribute _interpolation_func."""
+        interpolation = INTERPOLATION_TYPES[self.__interpolation__]
+        if interpolation == 0:  # linear
+
+            def linear_interpolation(x, x_min, x_max, x_data, y_data, coeffs):
+                x_interval = bisect_left(x_data, x)
+                x_left = x_data[x_interval - 1]
+                y_left = y_data[x_interval - 1]
+                dx = float(x_data[x_interval] - x_left)
+                dy = float(y_data[x_interval] - y_left)
```

Comment:
Do we really need to cast the calculations here? Can we just assume the inputs will always be numbers? I know it may not impact performance that much, but I would avoid redundant operations anyways.

```suggestion
                dx = x_data[x_interval] - x_left
                dy = y_data[x_interval] - y_left
```

Reply:
I fell like that is something added to avoid some typing problems with numpy. Changing this might create some bugs